src/connect_to_peers.rs

```diff
@@ -760,6 +780,53 @@ mod connect_tests {
         Ok(())
     }
 
+    #[traced_test]
+    #[tokio::test]
+    async fn bootstrap_node_refuses_reconnects_within_disconnect_cooldown_period() -> Result<()> {
+        let network = Network::Main;
+        let args = cli_args::Args {
+            bootstrap: true,
+            network,
+            ..Default::default()
+        };
+
+        let (_broadcast_tx, broadcast_rx, to_main_tx, _to_main_rx, mut state_lock, handshake) =
+            get_test_genesis_setup(network, 0, args).await?;
+        let node_0_address = get_dummy_socket_address(0);
+
+        state_lock
+            .lock_guard_mut()
+            .await
+            .net
+            .register_peer_disconnection(node_0_address, SystemTime::now());
```

Comment:
Consider adding a dual test case, or iterating over variants within this test case, to verify that the connection *is* allowed after the cooldown period has exceeded.

I highlight these lines because I think that's where you can change the effective elapsed time.

Reply:
I have expanded the existing test case.